### PR TITLE
#1 - Create full_coupler_mod 

### PR DESCRIPTION
**Pull request by mlee03**
In this PR,  `module use` statements and subroutine definitions found in `coupler_main.F90` have been moved to a newly created `full_coupler_mod` module.  In  this version of `full_coupler_mod`  most of the imported procedures have been marked as `public` but as coupler subroutines are defined, many of these procedures will become private.

Experiments in AM4, AM5, CM4, SPEAR, OM4_extra, and OMIP_CORE2 xmls for regression testing pass.

### CODE REVIEW

**Comment by thomas-robinson** [comment]
@mlee03 why not jus merge into main?  Why are you using a coupler_dev branch?

**Comment by mlee03** [comment]
@thomas-robinson I don't want to be the person who broke main :grin:

**Comment by thomas-robinson** [comment]
`main` is the development branch.  That's where this PR should go.  We don't need a different branch for everyone's development.

File: full/coupler_main.F90 [hunk 1]
```diff
@@ -333,253 +333,14 @@ program coupler_main
...
-     else if (arg == '--fredb_id') then
-        write (output_unit,*) TRIM(fredb_id)
-        stop
-     end if
-  END DO
-
-  if (arg_count .ge. 1) then
-    write (error_unit,*) 'Usage: '//TRIM(executable_name)//' [--fredb_id]'
-    stop 1
-  end if
```
Comment (thomas-robinson) [inline full/coupler_main.F90:579]: You need to add this to the description of this PR or you need to open a new PR that has a more general cleanup/removal of parts of the code

**Review by thomas-robinson — CHANGES_REQUESTED** [review]

File: full/coupler_main.F90 (hunk 1 above)
Comment (mlee03) [reply full/coupler_main.F90]: Added back in and ready for another round of review!

**Comment by mlee03** [comment]
@thomas-robinson @bensonr, any additional changes?

**Comment by mlee03** [comment]
@thomas-robinson , BEHOLD, everything passes!  READY FOR REVIEW

**Review by thomas-robinson — APPROVED** [review]

**Review by uramirez8707 — APPROVED** [review]

**Pull request merged**